File: benchmarks/cross/progs/mandel.lua
local W,H,MAXI = 200,200,60
local total = 0
for py=0,H-1 do
  local y0 = py*2.5/H - 1.25
  for px=0,W-1 do
    local x0 = px*3.0/W - 2.0
    local x,y,i = 0.0,0.0,0
    while i < MAXI do
      local xx,yy = x*x, y*y
      if xx+yy > 4.0 then i = MAXI
      else
        local xt = xx-yy+x0
        y = 2.0*x*y + y0
        x = xt
        i = i + 1
        total = total + 1
      end
    end
  end
end
print(total)

File: benchmarks/cross/progs/mandel.js
const W=200,H=200,MAXI=60;
let total=0;
for(let py=0;py<H;py++){
  const y0=py*2.5/H-1.25;
  for(let px=0;px<W;px++){
    const x0=px*3.0/W-2.0;
    let x=0.0,y=0.0,i=0;
    while(i<MAXI){
      const xx=x*x, yy=y*y;
      if(xx+yy>4.0){ i=MAXI; }
      else { const xt=xx-yy+x0; y=2.0*x*y+y0; x=xt; i++; total++; }
    }
  }
}
console.log(total);
